library/dokku_config.py

```diff
@@ -69,79 +85,159 @@ def subprocess_check_output(command, split='\n'):
 
 
 def dokku_config(app):
-    command = 'dokku config:export --format json {0}'.format(app)
-    output, error = subprocess_check_output(command, split=None)
+    if app == '--global':
+        command = 'dokku config:export --format json --global'
+        output, error = subprocess_check_output(command, split=None)
+    else:
+        command = 'dokku config:export --format json {0}'.format(app)
+        output, error = subprocess_check_output(command, split=None)
 
     if error is None:
         try:
             output = json.loads(output)
         except ValueError as e:
             error = str(e)
-
+    # error = output
     return output, error
 
 
 def dokku_config_set(data):
     is_error = True
     has_changed = False
     meta = {'present': False}
-
+    meta = {'error1': None}
+    options = []
+    
     values = []
     invalid_values = []
     existing, error = dokku_config(data['app'])
     for key, value in data['config'].items():
+        # if it's not a valid string or unicode string, add to invalid, skip
         if not isinstance(value, basestring):
             invalid_values.append(key)
             continue
-
+        # if the value of the envar is unchanged, skip
         if value == existing.get(key, None):
             continue
         values.append('{0}={1}'.format(key, pipes.quote(value)))
 
     if invalid_values:
         template = 'All values must be keys, found invalid types for {0}'
-        meta['error'] = template.format(', '.join(invalid_values))
+        meta['error1'] = template.format(', '.join(invalid_values))
         return (is_error, has_changed, meta)
 
     if len(values) == 0:
         is_error = False
         has_changed = False
         return (is_error, has_changed, meta)
-
-    command = 'dokku config:set {0} {1}'.format(data['app'], ' '.join(values))
+    
+    if not data['restart']:
+        options.append('--no-restart') 
+    
+    # you can further modify this to add other params such as "encoded" or other options in future dokku versions
+
+    if data['app'] == '--global':
+        command = 'dokku config:set {1} --global {0}'.format(' '.join(values), ' '.join(options))
+    else:
+        command = 'dokku config:set {2} {0} {1}'.format(data['app'], ' '.join(values), ' '.join(options))
```

Comment:
This whole block can just be:

```
command = 'dokku config:set {0} {1} {2}'.format(' '.join(options), data['app'], ' '.join(values))
```

Reply:
Yeah, good point. That originally applied until I realized the app could just be named “—global” and then I forgot to change it.